examples/autoqasm/6_Customize_gate_calibrations.ipynb

```diff
@@ -0,0 +1,391 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "bae7731a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "OPENQASM 3.0;\n",
+      "rx(pi / 2) $0;\n",
+      "rz(0.123) $1;\n",
+      "bit __bit_0__;\n",
+      "__bit_0__ = measure $0;\n"
+     ]
+    }
+   ],
+   "source": [
+    "@aq.main\n",
+    "def my_program():\n",
+    "    rx(\"$0\", np.pi/2)\n",
+    "    rz(\"$1\", 0.123)\n",
+    "    measure(\"$0\")\n",
+    "\n",
+    "main_program = my_program()\n",
+    "print(main_program.to_ir())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c7579516",
+   "metadata": {},
+   "source": [
+    "To attach gate calibrations to the program, call the `with_calibrations` method on `main_program` to create a new program with `my_rx_cal` attached, leaving the original intact. This allows you to experiment with different gate calibrations on the same program without the need to redefine the main program every time. \n",
+    "\n",
+    "In the printed OpenQASM script of the program, the gate definition for the `rx` gate becomes a `defcal` block."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "6b7cc158",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "OPENQASM 3.0;\n",
+      "extern drag_gaussian(duration, duration, float[64], float[64], bool) -> waveform;\n",
+      "waveform wf_drag_gaussian_1 = drag_gaussian(24.0ns, 2.547965400864ns, 2.370235498840002e-10, 0.293350447987059, false);\n",
+      "defcal rx(pi / 2) $0 {\n",
+      "    barrier $0;\n",
+      "    shift_frequency(q0_rf_frame, -321147.14178613486);\n",
+      "    play(q0_rf_frame, wf_drag_gaussian_1);\n",
+      "    shift_frequency(q0_rf_frame, 321147.14178613486);\n",
+      "    barrier $0;\n",
+      "}\n",
+      "rx(pi / 2) $0;\n",
+      "rz(0.123) $1;\n",
+      "bit __bit_0__;\n",
+      "__bit_0__ = measure $0;\n"
+     ]
+    }
+   ],
+   "source": [
+    "custom_program = main_program.with_calibrations(my_rx_cal)\n",
+    "print(custom_program.to_ir())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f038b8c8",
+   "metadata": {},
+   "source": [
+    "You can also define a gate calibration with variable parameters. Variable parameters are used in the body of the decorated function. So, the variable parameters must be arguments of the decorated function to form a valid Python function, instead of being the arguments the decorator `@aq.gate_calibration`. The variable parameters must have a type hint. The type hint for qubits is `aq.Qubit`, and the type hint for angles is `float`. Let's define another gate calibration with a variable parameter. This time, we define a calibration for the `rz` gate on qubit `$1` and a variable angle `angle`."
```

Comment:
```suggestion
    "You can also define a gate calibration with variable parameters. Variable parameters are used in the body of the decorated function. The variable parameters must be arguments of the decorated Python function, instead of being arguments to the decorator `@aq.gate_calibration`. The variable parameters must have a type hint of either `aq.Qubit` for qubits or `float` for angles. Let's define another gate calibration with a variable parameter. This time, we define a calibration for the `rz` gate on qubit `$1` and a variable angle `angle`."
```